PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskEmcalJetPerformance.cxx

```diff
@@ -1301,6 +1322,70 @@ void AliAnalysisTaskEmcalJetPerformance::DoTriggerSimulation()
   fHistManager.FillTH2(histname.Data(), nBkgPatchesEMCal, kEMCal);
   fHistManager.FillTH2(histname.Data(), nBkgPatchesDCal, kDCal);
   
+  // Fill max patch response and total EMCal energy response, if requested
+  if (fGeneratorLevel && fDoTriggerResponse && maxPatch) {
+    
+    Double_t phiMinMaxPatch = maxPatch->GetPhiMin();
+    Double_t phiMaxMaxPatch = maxPatch->GetPhiMax();
+    Double_t etaMinMaxPatch = maxPatch->GetEtaMin();
+    Double_t etaMaxMaxPatch = maxPatch->GetEtaMax();
+    Double_t truthEnergyPatch = 0;
+    
+    Double_t phiMinEMCal = fGeom->GetArm1PhiMin() * TMath::DegToRad(); // 80
+    Double_t phiMaxEMCal = fGeom->GetEMCALPhiMax() * TMath::DegToRad(); // ~188
+    Double_t detEnergyEMCal = 0;
+    Double_t truthEnergyEMCal = 0;
+    
+    AliTLorentzVector part;
+    Double_t partEta;
+    Double_t partPhi;
+    Double_t partE;
+    for (auto mcpart:fGeneratorLevel->accepted_momentum()) {
+      part.Clear();
+      part = mcpart.first;
+      partEta = part.Eta();
+      partPhi = part.Phi_0_2pi();
+      partE = part.E();
```

Comment:
From what I understand you sum up the energy of all particles including hadrons. Hadrons however don't deposit the full energy in the EMCAL. You either have to apply an effective hadronic correction similar to the correction applied in the hadronic correction component or you take the parameterized response from Mateusz. Only for electrons and photons you should assume full energy deposit in the EMCAL. Note this is only relevant at generator level, at cluster level the detector response has already been applied.